### PR TITLE
Fixes #1190: Remove empty anchor elements

### LINKED ISSUE #1190: Remove empty anchor elements
Some blog platforms seem to inject heading anchors with no inner content:

Screen Shot 2020-10-12 at 5 01 30 PM

This causes the content to shift to the right by 43px, and throws off alignment.  We could strip out these empty `<a></a>` elements from the DOM in our post-processing.

### PR DESCRIPTION
**Pull request by tonyvugithub**
## Issue This PR Addresses

Fixed #1190 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Added exclusiveFilter option to sanitizeHtml function in sanitize.js . The option removed anchor tags with no content in all headings.

**Before**
image
***
**After**
image

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/backend/utils/html/sanitize.js [hunk 1]
```diff
@@ -47,5 +47,8 @@ module.exports = function (dirty) {
       a: ['href'],
     },
     allowedIframeHostnames: ['www.youtube.com', 'player.vimeo.com'],
+    exclusiveFilter: (frame) => {
```
Comment (humphd) [inline src/backend/utils/html/sanitize.js:50]: This needs to go in a different PR vs. riding along with this other fix.

**Review by humphd — CHANGES_REQUESTED** [review]
This looks good, but needs a test.  You need to write something very similar to https://github.com/Seneca-CDOT/telescope/blob/master/test/replace-entities.test.js, but you would test cases like:

1. `<a></a>` should get removed
1. `<a></a><a></a><a>should still exist</a>` should only remove third anchor
1. `<a>foo</a>` should not get removed
1. `<a href="http://localhost">localhost</a>` should not get removed
1. `<a href="#abc">abc</a>` should not get removed

etc.

**Comment by tonyvugithub** [comment]
@humphd : I have updated the tests. Ready to merge

**Comment by tonyvugithub** [comment]
@humphd CI does not like my loop in one of the test, be right back

**Review by humphd — APPROVED** [review]

**Review by cindyledev — APPROVED** [review]

**Comment by humphd** [comment]
@tonyvugithub can you rebase this onto master and squash this for me and push a new commit?

**Comment by tonyvugithub** [comment]
@humphd : I have not watched the video for this week yet. Can I do it later tonight?

**Comment by humphd** [comment]
@tonyvugithub sure, no rush.  Let us know if you need help.

**Pull request closed without merging**